semanticdb/semanticdb3/guide.md

```diff
@@ -0,0 +1,576 @@
+    says that `main` is a method with one parameter of type `Array[String]`.
+    Further lines of the printout establish that `Array` in that type
+    refers to `scala.Array#`, etc.
+  * `Occurrences` contains a list of identifiers from the source file with
+    their line/column-based positions and unique identifiers pointing to
+    corresponding definitions resolved by the compiler.
+
+    For example, `[2:4..2:11): println => scala.Predef.println(Any).` says that
+    the identifier `println` on line 3 (zero-based numbering scheme!) refers
+    to the one-argument `println` overload from `scala.Predef`.
+  * For more information on various parts of SemanticDB payloads,
+    check out [the specification](semanticdb3.md).
+
+## What is SemanticDB good for?
+
+SemanticDB decouples producers and consumers of semantic information about
+programs and establishes a rigorous specification of the interchange
+format.
+
+Thanks to that, SemanticDB-based tools like [Scalafix](https://github.com/scalacenter/scalafix),
+[Metadoc](https://github.com/scalameta/metadoc)
+and [Metals](https://github.com/scalameta/metals) don't need to know about compiler
+internals and can work with any compiler that supports SemanticDB.
+This demonstrably improves developer experience, portability and scalability.
+Next-generation semantic tools at Twitter are based on SemanticDB.
+
+For more information about the SemanticDB vision, check out our talks:
+  * [Semantic Tooling at Twitter](https://www.youtube.com/watch?v=4yqDFsdKciA)
+    (June 2017) by Eugene Burmako & Stu Hood.
+  * [SemanticDB for Scala developer tools](https://t.co/Z080F6JxyQ?amp=1)
+    (April 2018) by Ólafur Páll Geirsson.
+  * [How We Built Tools That Scale to Millions of Lines of Code](https://na.scaladays.org/schedule/how-we-built-tools-that-scale-to-millions-of-lines-of-code)
+    (June 2018) by Eugene Burmako.
+
+## Producing SemanticDB
+
+### Scalac compiler plugin
+
+Scalameta includes the `semanticdb-scalac` compiler plugin.
+This plugin injects itself immediately after the `typer` phase of the Scala
+compiler and then harvests and dumps semantic information from Scalac
+in SemanticDB format. The plugin supports the following options that can
+be passed through Scalac:
+
+<table>
+  <tr>
+    <td width="250px">Option</td>
+    <td>Value</td>
+    <td>Explanation</td>
+    <td>Default</td>
+  </tr>
+  <tr>
+    <td><code>-P:semanticdb:sourceroot</code></td>
+    <td>Absolute or relative path</td>
+    <td>
+      Used to relativize source file paths into
+      <code>TextDocument.uri</code>.
+    </td>
+    <td>Current directory</td>
+  </tr>
+  <tr>
+    <td><code>-P:semanticdb:targetroot</code></td>
+    <td>Absolute or relative path</td>
+    <td>Designates a directory that will contain generated SemanticDB files.</td>
+    <td>Output directory of the Scala compiler</td>
+  </tr>
+  <tr>
+    <td><code>-P:semanticdb:mode</code></td>
+    <td>Enumeration</td>
```

Comment:
Instead of `Enumeration`, let's list the valid options like `all | definitions | none`